### PR TITLE
OpenSSL 1.1.0 compatibility

### PR DESCRIPTION
**Pull request by nouzun**
Changes to be able to build librdkafka with OpenSSL 1.1.0

### CODE REVIEW

File: src/rdkafka_transport.c [hunk 1]
```diff
@@ -503,6 +507,13 @@ void rd_kafka_transport_ssl_init (void) {
 	SSL_load_error_strings();
 	SSL_library_init();
 	OpenSSL_add_all_algorithms();
+#else
+#ifdef HAVE_OPENSSL_CRYPTO_THREADID_SET_CALLBACK
+	CRYPTO_THREADID_set_callback(rd_kafka_transport_ssl_threadid_cb);
```
Comment (edenhill) [inline src/rdkafka_transport.c]: Can't find if this is actually needed on OpenSSL >=1.1.0, but other projects seem to not use it on that version, so maybe it should be moved to the < 1.1.0 block?

https://github.com/pycurl/pycurl/issues/491

File: src/rdkafka_transport.c [hunk 2]
```diff
@@ -503,6 +507,13 @@ void rd_kafka_transport_ssl_init (void) {
...
+#endif
+	OPENSSL_init_ssl(0, NULL);
```
Comment (edenhill) [inline src/rdkafka_transport.c]: This isn't needed according to docs:
https://www.openssl.org/docs/man1.1.0/ssl/OPENSSL_init_ssl.html

Comment (nouzun) [reply src/rdkafka_transport.c]: I removed it.

File: src/rdkafka_transport.c (hunk 1 above)
Comment (nouzun) [reply src/rdkafka_transport.c]: True. I moved it to the < 1.1.0 block.

**Pull request merged**